packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -14,45 +13,114 @@ import { createPopperScope } from '@radix-ui/react-popper';
 import { Portal } from '@radix-ui/react-portal';
 import { Slottable } from '@radix-ui/react-slot';
 import * as VisuallyHiddenPrimitive from '@radix-ui/react-visually-hidden';
-import { useCallbackRef } from '@radix-ui/react-use-callback-ref';
 import { useId } from '@radix-ui/react-id';
-import { createStateMachine } from './createStateMachine';
-import { tooltipStateChart } from './tooltipStateChart';
 
 import type * as Radix from '@radix-ui/react-primitive';
 import type { Scope } from '@radix-ui/react-context';
 
 /* -------------------------------------------------------------------------------------------------
- * State machine
+ * TooltipProvider
  * -----------------------------------------------------------------------------------------------*/
 
-type StateAttribute = 'closed' | 'delayed-open' | 'instant-open';
-const stateMachine = createStateMachine(tooltipStateChart);
+const PROVIDER_NAME = 'TooltipProvider';
+const DEFAULT_DELAY_DURATION = 700;
+const TOOLTIP_OPEN = 'tooltip.open';
+
+type ScopedProps<P = {}> = P & { __scopeTooltipProvider?: Scope };
+const [createTooltipContext, createTooltipScope] = createContextScope(PROVIDER_NAME, [
+  createPopperScope,
+]);
+const usePopperScope = createPopperScope();
+
+type TooltipProviderContextValue = {
+  isOpenDelayed: boolean;
+  delayDuration: number;
+  onTooltipOpen(): void;
+  onTooltipClose(): void;
+};
+
+const [TooltipProviderProvider, useTooltipProviderContext] =
+  createTooltipContext<TooltipProviderContextValue>(PROVIDER_NAME, {
+    isOpenDelayed: true,
+    delayDuration: DEFAULT_DELAY_DURATION,
+    onTooltipOpen: () => {},
+    onTooltipClose: () => {},
+  });
+
+interface TooltipProviderProps {
+  /**
+   * The duration from when the mouse enters the trigger until the tooltip gets opened.
+   * @defaultValue 700
+   */
+  delayDuration?: number;
+  /**
+   * How much time a user has to enter another trigger without incurring a delay again.
+   * @defaultValue 300
+   */
+  skipDelayDuration?: number;
```

Comment:
I copied these props to the provider and removed `skipDelayDuration` from each `Tooltip` because it only makes sense in the context of *multiple* tooltips.

The provider `delayDuration` will be the default but each tooltip can override it with their own.

Reply:
Allowing different delays/durations/thresholds was intended. We've seen a bunch of products that have a delay on some tooltips but not delay on others.

Reply:
Ah ok nice, I didn't know. So at least for the opening delay we can still control that per tooltip. Do you think the other is ok as a global? (ie. the time allowed between 2 tooltips)

Reply:
I think so. I'd even bet it's ok hardcoded.